README.md

```diff
@@ -1,10 +1,177 @@
+
+[Adafruit ItsyBitsy M4 Express]: https://www.adafruit.com/product/3800
+
+### Running and debugging the experiments
+
+This repository uses [`probe-run`] to enable running (and to some extent
+debugging) embedded applications just like native ones using `cargo run`. This
+includes support for highly performant stdin/stdout and automatic panic message
+output using the [Real-Time Transfer (RTT)] I/O protocol from the [`probe-rs`]
+project. For more serious debugging a configuration for [`cargo-embed`] is
+provided, which will provide a GDB server when run with `cargo embed debug
+<args>` which can then be attached to using a multi-arch GDB build either from
+the CLI or your favorite IDE.
+
+[`probe-rs`]: https://probe.rs/
+
+[`probe-run`]: https://github.com/knurling-rs/probe-run
+
+[`cargo-embed`]: https://github.com/probe-rs/cargo-embed
+
+[Real-Time Transfer (RTT)]: https://github.com/probe-rs/probe-rs-rtt
+
+### Debug Probe needed
+
+The use of `probe-rs`-based tooling (`probe-run` and `cargo-embed`) requires you
+to attach to your MCU board using a [debug probe] supporting the debugging
+protocol of your particular MCU (either [JTAG] or preferably [SWD] for ARM
+Cortex-M based chips like the [ATSAMD51G19A]).
+
+In addition to considering what protocol is used between the MCU and the debug
+probe, it's also important to check what protocol the **debug probe uses to talk
+to the host computer** (the machine where you're uploading code and debugging
+from). At present (August 23rd, 2021) the `probe-rs` [supported probe protocols]
+include CMSIS-DAP, ST-LINK, J-Link and experimentally FTDI (JTAG only). Feel
+free to look into each to understand their differences and use what works for
+you, but detailed below is the low cost and quite versatile setup that Racklet
+core developers have confirmed to be working.
+
+[debug probe]: https://developer.arm.com/tools-and-software/embedded/debug-probes
+
+[JTAG]: https://en.wikipedia.org/wiki/JTAG
+
+[SWD]: https://developer.arm.com/architectures/cpu-architecture/debug-visibility-and-trace/coresight-architecture/serial-wire-debug
+
+[supported probe protocols]: https://github.com/probe-rs/probe-rs/tree/master/probe-rs/src/probe
+
+### ST-LINK V2 clone as a CMSIS-DAP adapter
+
+There are various low-cost [clones] of the ST-LINK V2 debugger probe available
+from markets such as eBay and Aliexpress. It doesn't really matter which one you
+choose/have (the original one should work as well), as the stock firmware that
+only speaks the ST-LINK protocol needs to be replaced. While it could
+technically work with `probe-rs`, the firmware is proprietary and has issues
+talking to anything other than STM32 ARM chips. Our recommendation is flashing
+the probe with the [dap42] CMSIS-DAP firmware, which enable the probe to support
+basically any ARM chip (including STM32) and to work seamlessly with `probe-rs`
+(well once [#767] has been merged anyways, use [my `probe-rs` fork] in the
```

Comment:
@twelho FYI it has already be merged 🎉

Reply:
Sorry, meant to say that it needs to make its way in to a release so that the other tools and projects pulling in `probe-rs` can utilize it. I've forked [probe-run](https://github.com/twelho/probe-run), [cargo-embed](https://github.com/twelho/cargo-embed) and [cargo-flash](https://github.com/twelho/cargo-flash) to apply that fix until the next `probe-rs` release happens. Fixed in https://github.com/racklet/bmc-experiments/pull/3/commits/fdbd1d8795195885e03189526b7527b3f51b64b7.